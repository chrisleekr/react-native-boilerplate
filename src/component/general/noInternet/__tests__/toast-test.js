import 'react-native';
import React from 'react';

import {cleanup, render} from '@testing-library/react-native';

let mockIsConnected = true;

jest.mock('@react-native-community/netinfo', () => ({
  useNetInfo: () => ({
    isConnected: mockIsConnected
  })
}));

import NoInternetToast from '../toast';

let rendered;

describe('NoInternetToast', () => {
  afterEach(cleanup);

  describe('when connected', () => {
    beforeEach(() => {
      mockIsConnected = true;
      rendered = render(<NoInternetToast />);
    });

    it('matches snapshot', () => {
      expect(rendered.toJSON()).toMatchSnapshot();
    });

    it('does not show error message', () => {
      expect(rendered.queryByTestId('text')).toBeFalsy();
    });
  });

  describe('when is not connected', () => {
    beforeEach(() => {
      mockIsConnected = false;
      rendered = render(<NoInternetToast />);
    });

    it('matches snapshot', () => {
      expect(rendered.toJSON()).toMatchSnapshot();
    });

    it('shows error message', () => {
      expect(rendered.queryByTestId('text')).toBeTruthy();
    });
  });
});
