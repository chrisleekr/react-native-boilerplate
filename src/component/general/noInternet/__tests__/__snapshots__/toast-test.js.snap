// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoInternetToast when connected matches snapshot 1`] = `null`;

exports[`NoInternetToast when is not connected matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d70015",
      "bottom": 0,
      "flexDirection": "row",
      "height": 18,
      "justifyContent": "center",
      "position": "absolute",
      "width": 750,
      "zIndex": 10,
    }
  }
  testID="no-internet-toast"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 11,
      }
    }
    testID="text"
  >
    No Internet Connection
  </Text>
</View>
`;
